### PR TITLE
Adding a safe toNonEmpty

### PR DESCRIPTION
**Pull request by nhibberd**
Basically safe toList for NonEmpty

### CODE REVIEW

**Comment by erikd-ambiata** [comment]
:+1: on :green_apple:

File: src/P/List.hs [hunk 1]
```diff
@@ -45,3 +46,11 @@ lastMaybe = listToMaybe . reverse
...
 count :: (a -> Bool) -> [a] -> Int
 count predicate = length . filter predicate
+
+toNonEmpty :: [a] -> Maybe (NonEmpty a)
+toNonEmpty l =
+  case l of
+    h : t ->
+      Just (h :| t)
+    _ ->
+      Nothing
```
Comment (domdere) [inline src/P/List.hs]: [this one](https://hackage.haskell.org/package/semigroups-0.18.1/docs/Data-List-NonEmpty.html#v:nonEmpty) no good?

Comment (jystic) [inline src/P/List.hs]: yeah would be better to re-export that imo

**Comment by jystic** [comment]
👍

**Pull request merged**